File: content/src/main/java/org/sandboxpowered/silica/minecraft/blocks/RedstoneWireBlock.java
package org.sandboxpowered.silica.minecraft.blocks;

import org.sandboxpowered.api.block.BaseBlock;
import org.sandboxpowered.api.block.Block;
import org.sandboxpowered.api.state.BlockState;
import org.sandboxpowered.api.state.Properties;
import org.sandboxpowered.api.state.StateFactory;
import org.sandboxpowered.api.state.property.EnumProperty;

public class RedstoneWireBlock extends BaseBlock {
    private static final EnumProperty<Side> NORTH = EnumProperty.of("north", Side.class);
    private static final EnumProperty<Side> EAST = EnumProperty.of("east", Side.class);
    private static final EnumProperty<Side> SOUTH = EnumProperty.of("south", Side.class);
    private static final EnumProperty<Side> WEST = EnumProperty.of("west", Side.class);

    public RedstoneWireBlock(Settings settings) {
        super(settings);
    }

    @Override
    public void appendProperties(StateFactory.Builder<Block, BlockState> builder) {
        super.appendProperties(builder);
        builder.add(Properties.POWER, NORTH, EAST, SOUTH, WEST);
    }

    public static enum Side {
        UP,
        SIDE,
        NONE
    }
}

File: modules/vanilla/common/src/main/kotlin/org/sandboxpowered/silica/vanilla/block/RedstoneWireBlock.kt
package org.sandboxpowered.silica.vanilla.block

import org.sandboxpowered.silica.api.block.BaseBlock
import org.sandboxpowered.silica.api.block.Block
import org.sandboxpowered.silica.api.Identifier
import org.sandboxpowered.silica.api.world.state.StateProvider
import org.sandboxpowered.silica.api.world.state.block.BlockState
import org.sandboxpowered.silica.vanilla.block.BlockProperties.POWER
import org.sandboxpowered.silica.vanilla.block.BlockProperties.REDSTONE_SIDE_EAST
import org.sandboxpowered.silica.vanilla.block.BlockProperties.REDSTONE_SIDE_NORTH
import org.sandboxpowered.silica.vanilla.block.BlockProperties.REDSTONE_SIDE_SOUTH
import org.sandboxpowered.silica.vanilla.block.BlockProperties.REDSTONE_SIDE_WEST

class RedstoneWireBlock(identifier: Identifier) : BaseBlock(identifier) {
    override fun appendProperties(builder: StateProvider.Builder<Block, BlockState>) {
        super.appendProperties(builder)
        builder.add(REDSTONE_SIDE_NORTH, REDSTONE_SIDE_EAST, REDSTONE_SIDE_SOUTH, REDSTONE_SIDE_WEST, POWER)
    }
}